docs/pages/guides/adding-tokens.mdx

```diff
@@ -0,0 +1,681 @@
+       address tokenAddress = vm.envAddress("TOKEN_ADDRESS");
+   ```
+
+   Read our configuration from `.env`.
+
+   ```solidity
+       // Specify a store so that you can use tables directly
+       StoreSwitch.setStoreAddress(worldAddress);
+
+       // Start broadcasting transactions from the deployer account
+       vm.startBroadcast(privateKey);
+
+       // Register the singleton tables
+       Token.register();
+       RewardSize.register();
+   ```
+
+   Create the two singleton tables for our state variables.
+
+   ```solidity
+       // Set the singleton values so they'll be available to RewardSystem
+       Token.set(tokenAddress);
+       RewardSize.set(10 ** 18);
+   ```
+
+   Specify the value.
+   `10**18` is 10<sup>18</sup>, so one token (we use the same 18 digits standard as ETH).
+
+   ```solidity
+       IWorld world = IWorld(worldAddress);
+       ResourceId systemResource = WorldResourceIdLib.encode(RESOURCE_SYSTEM, "DDDD", "RewardSystem");
+
+       RewardSystem rewardSystem = new RewardSystem();
+       console.log("RewardSystem address: ", address(rewardSystem));
+   ```
+
+   Deploy the `RewardSystem` contract.
+
+   ```solidity
+       world.registerSystem(systemResource, rewardSystem, false);
+   ```
+
+   Register `RewardSystem`.
+   Note that the third parameter is `false`.
+   We do not allow unauthorized entities to call `giveReward` because otherwise anybody could reward themselves, rendering the token meaningless.
+
+   ```solidity
+       world.registerFunctionSelector(systemResource, "giveReward(address)");
+   ```
+
+   Register the function selector.
+   Because this function is in a non-root namespace, it gets a fully qualified name, `DDDD_RewardSystem_giveReward`, when called through the `World`.
+
+   ```solidity
+       // Allow systems in the root namespace to access rewardSystem.
+       world.grantAccess(systemResource, worldAddress);
+   ```
+
+   We need `TaskSystem`, which is in the root namespace, to be able to give rewards.
+   Because it is a root namespace `System`, it runs at the `World` address so that's the address that needs the access.
+
+   ```solidity
+       // Mint a million tokens for rewardSystem.
+       IERC20Mintable token = IERC20Mintable(tokenAddress);
+       token.mint(address(rewardSystem), 10 ** 6 * RewardSize.get());
+   ```
+
+   Normally the only entity allowed to mint ERC-20 tokens is the namespace owner.
+   The easiest way to make sure `RewardSystem` has enough tokens to give is simply to give it enough tokens, and refill as needed.
+
+   ```solidity
+       // This is a workaround for issue
+       // https://github.com/latticexyz/mud/issues/2089
+       token.mint(address(worldAddress), 10 ** 6 * RewardSize.get());
+   ```
+
+   The issue is that even though `RewardSystem` is in a non-root namespace (`DDDD`), when called from `TaskSystem` it runs as the `World` address - so we need to give that address tokens to distribute.
```

Comment:
i'd prefer to not have this in the docs, let's wait for the issue to be fixed. Alternatively we can already merge the first part of the guide and hold back on the second part (controlling from a system). Or yet another alternative would be to use a root system for now so the behavior of transferring to the world is expected.

Reply:
Any idea how long it will be until the bug is fixed? Do you want me to try and fix it?

Reply:
it's fixed now!